python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,423 @@
+        self._name = new_name
+
+    @property
+    def parent(self):
+        """The item's parent item."""
+        return self._graph.get_parent(self)
+
+    @property
+    def persistent(self):
+        """The item should not be removed when the graph is cleared."""
+        return self._persistent
+
+    @persistent.setter
+    def persistent(self, is_persistent):
+        """Set the item to persistent or not.
+
+        Only top-level items can be set to persistent.
+        """
+
+        if self.parent != self._graph.root_item:
+            raise sgtk.TankError(
+                "Only top-level graph items can be made persistent.")
+
+        self._persistent = is_persistent
+
+    @property
+    def properties(self):
+        """
+        A :class:`PublishData` instance where arbitrary data can be stored on
+        the item. The property itself is read-only (you can't assign a different
+        ``PublishData`` instance.
+
+        This property provides a way to store data that is global across all
+        attached publish plugins. It is also useful for accessing data stored
+        on parent items that may be useful to plugin attached to child items.
+
+        For properties that are local to the current plugin, see
+        ``local_properties``.
+
+        This property can also be used to store data on an items that may then
+        be accessed by plugins attached to the item's children.
+        """
+        return self._global_properties
+
+    @property
+    def tasks(self):
+        return self._graph.get_tasks(self)
+
+    @property
+    def type_spec(self):
+        """
+        The type specification for this item. This specification follows a
+        hierarchical dot notation. For example, 'file', 'file.image',
+        'file.quicktime' or 'maya_scene'. This specification is up to the client
+        code to define in terms of formatting and meaning.
+        """
+        return self._type_spec
+
+    @property
+    def type_display(self):
+        """
+        The display string for this item's type.
+        """
+        return self._type_display
+
+    def _get_local_properties(self):
+        """
+        Return properties local to the currently executing publish plugin.
+
+        This is done in a separate method to allow access to any method in this
+        class. We look 2 levels up in the stack to get the calling plugin class.
+        If this is called more than one level deep in this class, expect issues.
+        """
+
+        # try to determine the current publish plugin
+        calling_object = inspect.stack()[2][0].f_locals.get("self")
+
+        if not calling_object or not isinstance(calling_object, sgtk.hook.Hook):
+            raise AttributeError(
+                "Could not determine the current publish plugin when accessing "
+                "an item's local properties. Item: %s" % (self,))
+
+        plugin_id = calling_object.id
```

Comment:
We used to cache the object itself as the key to the lookup. The id property was added to plugins and used here so that items could be serialized/deserialized.